src/sentry/search/events/builder/metrics.py

```diff
@@ -595,42 +597,42 @@ def validate_orderby_clause(self) -> None:
             ) or (isinstance(orderby.exp, Function) and orderby.exp.alias == "title"):
                 raise IncompatibleMetricsQuery("Can't orderby tags")
 
+    def _run_query_with_metrics_layer(self, query: Query, is_alerts_query: bool = False) -> Any:
+        from sentry.snuba.metrics.datasource import get_series
+        from sentry.snuba.metrics.mqb_query_transformer import transform_mqb_query_to_metrics_query
+
+        try:
+            with sentry_sdk.start_span(op="metric_layer", description="transform_query"):
+                metric_query = transform_mqb_query_to_metrics_query(query, is_alerts_query)
+            with sentry_sdk.start_span(op="metric_layer", description="run_query"):
+                metrics_data = get_series(
+                    projects=self.params.projects,
+                    metrics_query=metric_query,
+                    use_case_id=UseCaseKey.PERFORMANCE
+                    if self.is_performance
+                    else UseCaseKey.RELEASE_HEALTH,
+                    include_meta=True,
+                )
+        except Exception as err:
+            raise IncompatibleMetricsQuery(err)
+        with sentry_sdk.start_span(op="metric_layer", description="transform_results"):
+            # series does some strange stuff to the clickhouse response, turn it back so we can handle it
+            metric_layer_result: Any = {
+                "data": [],
+                "meta": metrics_data["meta"],
+            }
+            for group in metrics_data["groups"]:
+                data = group["by"]
+                data.update(group["totals"])
+                metric_layer_result["data"].append(data)
+                for meta in metric_layer_result["meta"]:
+                    if data[meta["name"]] is None:
+                        data[meta["name"]] = self.get_default_value(meta["type"])
+            return metric_layer_result
```

Comment:
I'd rather we didn't move this to a helper function, since eventually run_query will just be calling this code directly once we're 100% on the metrics_layer

Reply:
This has been explicitly done because in the `AlertMetricsQueryBuilder.run_query()` method I have to get the snql for the transformer and run it with the metrics layer. If I would have called `super.run_query()` directly what would have happened is that via the MRO (https://github.com/getsentry/sentry/blob/426295d4aee62db35c2331f23c5b5e1e2583cd90/src/sentry/search/events/builder/metrics.py#L634-L636) we would have had the snql being generated by the `AlertMetricsQueryBuilder.get_snql_query()` but that snql is already out of `SnubaQueryBuilder`, thus it is not understood by the transformer.

TL;DR
I have done this in order to avoid duplication of the method.